### DISCUSSION TITLE
latex snippets in markdown

### DISCUSSION THREAD

**Discussion by Aceticia** [L3MON4D3/LuaSnip#630] [Q&A] [answered]
How to enable latex snippets in math blocks of markdown files? I'm trying to write latex in markdown but don't want to copy all the snippets from latex. For example, using the snippets from [here](https://github.com/iurimateus/luasnip-latex-snippets.nvim/tree/main/lua/luasnip-latex-snippets), how would one approach it?

**Comment by L3MON4D3** [comment, marked as answer]
Ah, luasnip gets the filetypes for the cursor-position from `ft_func` (set during `setup`), we ship a function to use treesitter to resolve the filetype at the cursor (`require("luasnip.extras.filetype_functions").from_pos_or_filetype`).
You'll also want to update `load_ft_func` to automatically lazy-load latex/tex-filetype for markdown-files, this can be done by passing
```lua
load_ft_func = require("luasnip.extras.filetype_functions").extend_load_ft({
	markdown = {"latex", "tex"}
}),
```
to `setup` as well.
